nodestream/pipeline/extractors/stores/splunk_extractor.py

```diff
@@ -0,0 +1,400 @@
+        return f"{self.base_url}/servicesNS/{self.user}/{self.app}/search/jobs"
+
+    def get_results_endpoint(self, search_id: str) -> str:
+        """Get the results endpoint for a specific search job."""
+        return f"{self.base_url}/servicesNS/{self.user}/{self.app}/search/jobs/{search_id}/results"
+
+    async def _create_search_job(self, client: AsyncClient) -> str:
+        """Create a search job using the official Splunk REST API pattern."""
+        job_data = {
+            "search": self._normalized_query,
+            "earliest_time": self.earliest_time,
+            "latest_time": self.latest_time,
+            "max_count": str(self.max_count),
+            "output_mode": "json",  # Request JSON format explicitly
+        }
+
+        self.logger.info(
+            "Creating Splunk search job",
+            extra={
+                "query": self._normalized_query,
+                "earliest_time": self.earliest_time,
+                "latest_time": self.latest_time,
+                "max_count": self.max_count,
+            },
+        )
+
+        response = await client.post(
+            self.get_jobs_endpoint(),
+            data=job_data,
+            headers={
+                **self._headers,
+                "Content-Type": "application/x-www-form-urlencoded",
+            },
+            auth=self._auth,
+        )
+
+        if response.status_code != 201:
+            raise HTTPStatusError(
+                f"Failed to create search job: {response.status_code}",
+                request=response.request,
+                response=response,
+            )
+
+        # Extract SID from response - try JSON first since we requested it
+        search_id = None
+
+        try:
+            result = response.json()
+            search_id = result.get("sid")
+            if search_id:
+                self.logger.info(
+                    "Created search job successfully", extra={"search_id": search_id}
+                )
+                return search_id
+        except json.JSONDecodeError:
+            self.logger.debug("Job creation response not JSON, trying XML parsing")
+
+        # Fallback to XML parsing
+        import xml.etree.ElementTree as ET
+
+        try:
+            root = ET.fromstring(response.text)
+            sid_elem = root.find(".//sid")
+            if sid_elem is not None:
+                search_id = sid_elem.text
+        except Exception as e:
+            self.logger.error(
+                "Failed to parse job creation response",
+                extra={"error": str(e), "response": response.text[:500]},
+            )
+
+        if not search_id:
+            raise RuntimeError(
+                f"Failed to extract search ID from response: {response.text[:500]}"
+            )
+
+        self.logger.info(
+            "Created search job successfully", extra={"search_id": search_id}
+        )
+        return search_id
+
+    async def _wait_for_job_completion(
```

Comment:
My IDE flags this as having too much "cognitive complexity" (Aka too many nested branches and loops)

It would be good to refactor chunks of this into smaller functions to reduce the amount of brainpower required to figure out what's going on.